tests/installation/addon_products_sle.pm

```diff
@@ -133,20 +133,27 @@ sub run {
             if (match_has_tag("addon-betawarning-$addon") or match_has_tag("addon-license-$addon")) {
                 if (match_has_tag("addon-betawarning-$addon")) {
                     send_key "ret";
-                    assert_screen "addon-license-beta";
+                    if (sle_version_at_least('15') && !check_screen('addon-license-beta')) {
```

Comment:
It's more a technical question for me: is `!check_screen(...)` couldn't be replaced my `!match_has_tag(...)`?

Other than this, LGTM :)

Reply:
Probably not in this case:

I just checked testapi.pm and it says:
'''
  match_has_tag($tag);
Returns true if last matched needle has C<$tag> else return C<undef>.
'''
Then, I couldn't not use match_has_tag('addon-license-beta') here unless it is involved in line 131: 
my @tags = ('addon-products', "addon-betawarning-$addon", "addon-license-$addon", 'import-untrusted-gpg-key');

And 'addon-license-beta' couldn't be added into the tags since it only appears after "addon-betawarning-$addon" asserted.

So, seems !check_screen(...) couldn't be replaced my !match_has_tag(...) here.

Reply:
Correct, they are not the same. But please try to avoid check_screen with non-default timeout as it's wasting a lot of time. I think in this case no test code should be required at all but a workaround *needle* should be used to accept the screen when actually no beta is shown with a reference to the bug. In other case you could use another multi-tag assert_screen. The only other feasible alternative I see is something like a `wait_still_screen` after the assert_screen and a `check_screen 'addon-licence-beta', 0` afterwards

Reply:
Only a workaround needle without code change won't work here, because "send_key 'alt-a';" afterwords will be received as "Add a new addon" and fails the test.
See http://openqa-apac1.suse.de/tests/146#step/addon_products_sle/25